ci/test-checks.sh

```diff
@@ -65,11 +65,25 @@ fi
 
  _ ci/order-crates-for-publishing.py
 
+nightly_clippy_allows=(
+  # This lint occurs all over the code base
+  "--allow=clippy::significant_drop_in_scrutinee"
+
+  # The prost crate, used by solana-storage-proto, generates Rust source that
+  # triggers this lint. Need to resolve upstream in prost
+  "--allow=clippy::derive_partial_eq_without_eq"
+
+  # This link seems to incorrectly trigger in
+  # `programs/bpf_loader/src/syscalls/{lib,cpi}.rs`
+  "--allow=clippy::explicit-auto-deref"
```

Comment:
nittiest of nits: for consistency, let's have this be 
```suggestion
  "--allow=clippy::explicit_auto_deref"
```

Reply:
Done